ui/src/settings.vue

```diff
@@ -317,7 +346,39 @@ export default {
                 this.$router.push("/projects");
             });            
         },
+        changePassword(e) {
+            e.preventDefault()
+            if(!this.validaterepeatPass || !this.validatePass) return;
+            this.$http.put(Vue.config.auth_api+"/local/setpass",{
+                password_old : this.form.currentPassword,
+                password: this.form.newPassword
+            }).then(res=>{
+                this.$bvToast.toast('Successfully Updated Password',{
+                    variant : "success",
+                    solid : true
+                });
+                this.$router.push(Vue.config.auth_signout);
+            }).catch(err=>{
+                this.$bvToast.toast('Error in updating password',{
+                    variant : "danger",
+                    solid : true
+                });
+                console.error(err);
+            });
+        }
     },
+
+    computed : {
+        validaterepeatPass() {
+            if(!this.form.repeatPassword || !this.form.newPassword) return;
+            return this.form.repeatPassword == this.form.newPassword;
+        },
+        validatePass() {
+            const regPattern = new RegExp("^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d)(?=.*[-+_!@#$%^&*.,?]).+$");
```

Comment:
Thank you for working on this. Do you think it's better to use something like this instead https://github.com/apertureless/vue-password-strength-meter#readme 
?

Instead of *preventing* user from using certain password, I like the approach of letting them pick whatever the password they'd like to use - but we inform them how strong the password is. Ultimately, we can't prevent the user from sharing the password with someone else or sending them via email, etc.. So requiring strong "password" only goes so far.. So I'd like our users to have freedom of using "weak" password as long as they are informed that they are using a weak password. If they still choose to use weak password, I assume that they have a good reason for that, and we should trust that there is a good reason for that.